### DISCUSSION TITLE
How do we expect users to understand exec_mem_limit ?

### DISCUSSION THREAD

**Discussion by dataroaring** [apache/doris#8455] [RFC]
exec_mem_limit is a session variable, which can be set by users. I think we should define it exactly to make users understood. For example , It is max memory consumption of a query on a be. If a query consumes memory beyond exec_mem_limit on  a be , it should be failed due to memory allocation. 

I  am not sure whether the above idea is acceptable.

Now, exec_mem_limit does not work because some memory allocation is not limited by it due to calling MemPool::allocate. Actually exec_mem_limit works at fragment instance level, it does not work at query level.  However, FragmentInstance is related to table, users can not expect how many fragment instances would run on a be, so it is difficult to make users understood.

Should we let exec_mem_limit limit memory consumption on a query on a be?

**Comment by xiaokang** [comment]
Maybe we can add a new session variable eg.query_memory_limit_bytes. If user set new variable query_memory_limit_bytes, the old exec_mem_limit variable is ignored. Otherwise, the old exec_mem_limit is still effective for compatibility.

**Comment by PRNDA** [comment]
哥们，你还是直接发中文吧.....

**Comment by 1032851561** [comment]
Deregister query/load memory tracker, queryId=e41b907bc4cf4169-91e9c665a54b1683, Limit=4.00 GB, CurrUsed=51.40 MB, PeakUsed=74.76 GB

exec_mem_limit=4G , query peak memory about 74G，exec_mem_limit does not work as expected.
